### PR TITLE
[Compute] `az vm extension list`: Add new parameter `--ids` to support listing extensions by VM id

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Related command**

`az vm extension list`

**Description**

az vm extension list to include support "--ids"
close https://github.com/Azure/azure-cli/issues/22527

**Testing Guide**

**History Notes**

[Compute] `az vm extension list`: Add new parameter `--ids` to support listing extensions by VM id

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 1]
```diff
@@ -1704,6 +1704,20 @@ def test_vm_extension_debian(self, resource_group):
...
+        self.kwargs.update({
+            'vm': self.create_random_name('vm-',15)
+        })
+
+        vm = self.cmd('vm create -n {vm} -g {rg} --image ubuntults').get_output_in_json()
+        self.kwargs.update({
+            'vm_id': vm['id']
+        })
+        self.cmd('vm extension list --ids {vm_id} ', checks=[
+            self.check('length([])', 0)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:1718]: Could we test cases with list results?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**